presto-hive/src/main/java/com/facebook/presto/hive/BackgroundHiveSplitLoader.java

```diff
@@ -551,6 +551,15 @@ private static Properties getPartitionSchema(Table table, Optional<Partition> pa
 
             int tableBucketCount = bucketHandle.get().getTableBucketCount();
             int readBucketCount = bucketHandle.get().getReadBucketCount();
+
+            if (tableBucketCount != readBucketCount && bucketFilter.isPresent()) {
```

Comment:
It looks like it can be done earlier (`HiveMetadata#getAlternativeLayoutHandle`). But it doesn't matter much, as this object is created when planning anyway.

Reply:
Yeah, I originally want to make it more like a execution limitation for now -- but it turns out still in planning. No strong preference though.